game-core/src/test/java/games/strategy/debug/error/reporting/ErrorReportComponentsTest.java

```diff
@@ -0,0 +1,86 @@
+package games.strategy.debug.error.reporting;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import swinglib.TestConstant;
+
+/**
+ * Ensure components are wired to the expected action handlers.
+ */
+@ExtendWith(MockitoExtension.class)
+class ErrorReportComponentsTest {
+
+  private ErrorReportComponents errorReportComponents = new ErrorReportComponents();
+
+  @Mock
+  private Consumer<UserErrorReport> errorReportHandler;
+
+  @Mock
+  private Supplier<UserErrorReport> guiReader;
+
+  @Mock
+  private UserErrorReport userErrorReport;
+
+  @Mock
+  private Runnable runnable;
+
+  @BeforeAll
+  static void disableSwingPopups() {
+    TestConstant.setTestConstant();
```

Comment:
No, funny enough. We are not setting a system prop, but a boolean value. That value means "we are in a headless test context." Once true, that should never change. 

Admittedly, this is not the cleanest approach to the problem. It would be great to be able to notify that it should turn off the headless checks and not attempt to render to the UI.